### ISSUE TITLE
[Spec]  Deconstruct DualScreen Nuget into Essentials and Maui.Core (Investigating)

### ISSUE THREAD

**Issue by PureWeen** [dotnet/maui#1164]
# Merge DualScreen nuget into Maui.Controls and Maui.Essentials

- TwoPaneView
  - This control requires no outside libraries. It inherits from Grid and just modifies grid properties based on device parameters. This control could easily just be merged into Maui.Controls as a main control people can use
- DualScreenInfo
  - All of the native APIs that back this class are now part of the Window AndroidX nuget so we can just move this into essentials

**Comment by conceptdev** [comment]
The [Xamarin.AndroidX.Window](https://www.nuget.org/packages/Xamarin.AndroidX.Window/) binding has been updated to match latest Google ([details here](https://github.com/xamarin/AndroidX/issues/325#issuecomment-882718035)), and the [Xamarin Surface Duo samples](https://github.com/microsoft/surface-duo-sdk-xamarin-samples/pull/16) are updated to use it.

For .NET MAUI, the ideal situation would be:

* Microsoft.Maui.Controls.DualScreen (namespace update to [Xamarin.Forms.DualScreen](https://www.nuget.org/packages/Xamarin.Forms.DualScreen/))
* If the AndroidX libraries are also having their namespace updated, create an updated version of Xamarin.AndroidX.Window for .NET MAUI
* Regarding the two points above: how do we get the existing layout and helper class migrated?
   * Put them in a newer version of Microsoft.Maui.Controls.DualScreen
   * Split them into other locations (eg. Essentials)

The first two bullets are probably the minimum we need to to ensure customers can continue to build dual-screen apps. The third bullet (and the topic of this issue) is a desirable update which would bring compatibility to a wider variety of devices such as foldables from other manufacturers.

**Comment by DamianSuess** [comment]
@PureWeen just checking, this performed in PR #5020, correct?

Asking because I'm eager to build upon the Foldable feature when it becomes available.  Thank you :)

**Comment by logikonline** [comment]
Can this be reconsidered before year's end?

**Comment by jfversluis** [comment]
@DamianSuess the NuGet should be available now, released as part of .NET 7.

@logikonline why is it of particular interest to you? The functionality is available in a NuGet right now, this only speaks of refactoring some code basically.

**Comment by DamianSuess** [comment]
@jfversluis you're the man! I'll be sure we'll get it into our next sprint for validation w/ our project.
I love Xamarin.Forms but I enjoy faster running apps with the latest supported C# even better 😉

**Comment by logikonline** [comment]
For the same reason @DamianSuess outlined, I moved a large project to Maui, but I am one of the outliers that have a Duo 2 and like to offer a better experience for those devices too.

If in your refactoring, can you also provide an option on the Shell to recognize the control being used on the page.  The reason for this, if a Duo is in full-screen mode (across both screens), the tabbar will try to center, and the middle tab is unusable.  I resolved this with a custom renderer on the old MasterDetail to change the gravity of the menu to favor a right or left align.

The Shell really should offer this to make the Duo usable imo.

**Comment by jfversluis** [comment]
Gotcha! But then, if I understand correctly @logikonline you're more interested in this functionality being available (which it is now) than this code being in a different place (what this issue is technically about)?

As for features, it's probably best to open something new if you want to see things added/changed. You already did mention though yourself that you are probably one of the outliers so I'm not sure how high it would rank in terms of priority. If this is something you'd really like, maybe it would be best to see if you can help us (and thus yourself) out by making one or more contributions.

**Comment by logikonline** [comment]
No. I'll continue to write the custom renderer for the Shell but I can't offer the functionality for the Duo since it is only available for XF. 

But if the individual bring the code refactor to Maui does the work, it would be nice if they provided the addition since it isn't in XF & is a known limitation.

I view Maui as an improvement in the experience not merely a translation.
